core/state_transition.go

```diff
@@ -307,9 +308,17 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	homestead := st.evm.ChainConfig().IsHomestead(st.evm.Context.BlockNumber)
 	istanbul := st.evm.ChainConfig().IsIstanbul(st.evm.Context.BlockNumber)
 	subnetEVM := st.evm.ChainConfig().IsSubnetEVM(st.evm.Context.Time)
+	isTxAllowList := st.evm.ChainConfig().IsTxAllowList(st.evm.Context.Time)
 
 	contractCreation := msg.To() == nil
 
+	if isTxAllowList {
```

Comment:
hm should we do this check before buying gas (i.e in `preCheck`)?

Reply:
Thoughts @aaronbuchwald?

Reply:
Ya let's move this into `preCheck` that's a much cleaner place to add this check. We should add to the comment above where `preCheck` is called within `TransitionDb()` as well.

To explain a bit:

We're still performing this check before we do any EVM execution takes place, so this doesn't present any DoS vulnerabilities where we have to throw something away after already doing work on it.

As for doing this before buying gas, I think the concern there would be that we return this error before modifying the staetdb in any way? For that, the error returned here means the entire transaction is invalid, not a VM related error, so the statedb won't be re-used with an incorrect modification made either way.

Reply:
I moved this, let me know if I did it correctly

Reply:
it seems good to me